src/main/java/uk/gov/justice/digital/job/DataHubJob.java

```diff
@@ -111,41 +121,46 @@ public void batchProcessor(JavaRDD<byte[]> batch) {
+                    String sourceName = tableInfo.getAs(SOURCE);
+                    String tableName = tableInfo.getAs(TABLE);
 
-                    val structuredLoadDataFrame = structuredZoneLoad.process(spark, dataFrameForTable, tableInfo);
-                    val structuredIncrementalDataFrame = structuredZoneCDC.process(spark, dataFrameForTable, tableInfo);
+                    val optionalSourceReference = sourceReferenceService.getSourceReference(sourceName, tableName);
 
-                    dataFrameForTable.unpersist();
+                    if (optionalSourceReference.isPresent()) {
+                        val sourceReference = optionalSourceReference.get();
 
-                    curatedZoneLoad.process(spark, structuredLoadDataFrame, tableInfo);
-                    val curatedCdcDataFrame = curatedZoneCDC.process(spark, structuredIncrementalDataFrame, tableInfo);
+                        rawZone.process(spark, dataFrameForTable, sourceReference);
 
-                    String sourceName = tableInfo.getAs(SOURCE);
-                    String tableName = tableInfo.getAs(TABLE);
+                        val structuredLoadDataFrame = structuredZoneLoad.process(spark, dataFrameForTable, sourceReference);
+                        val structuredIncrementalDataFrame = structuredZoneCDC.process(spark, dataFrameForTable, sourceReference);
 
-                    val optionalSourceReference = sourceReferenceService.getSourceReference(sourceName, tableName);
-                    optionalSourceReference.ifPresent(sourceReference -> {
-                                if (!curatedCdcDataFrame.isEmpty()) domainService
-                                        .refreshDomainUsingDataFrame(
-                                                spark,
-                                                curatedCdcDataFrame,
-                                                sourceReference.getSource(),
-                                                sourceReference.getTable()
-                                        );
-                            }
-                    );
+                        dataFrameForTable.unpersist();
+
+                        curatedZoneLoad.process(spark, structuredLoadDataFrame, sourceReference);
+                        val curatedCdcDataFrame = curatedZoneCDC.process(spark, structuredIncrementalDataFrame, sourceReference);
+
+                        if (!curatedCdcDataFrame.isEmpty()) domainService
+                                .refreshDomainUsingDataFrame(
+                                        spark,
+                                        curatedCdcDataFrame,
+                                        sourceReference.getSource(),
+                                        sourceReference.getTable()
+                                );
+                    } else {
+                        handleNoSchemaFound(spark, dataFrame, sourceName, tableName);
```

Comment:
When there is no schema found, the affected records are written here instead of within the structured zone.